### ISSUE TITLE
QuickFixGPS updates fail for some users

### ISSUE THREAD

**Issue by dlenski** [dlenski/ttblue#2]
After fixing #1, QFG update still fails for at least one user, with `errno` of `EAGAIN`.

```
$ ttblue -D -D # show write packet-by-packet
...
Updating QuickFixGPS...
  Saved 72 bytes to 00020001_20150827_221935.bin
  Downloading http://gpsquickfix.services.tomtom.com/fitness/sifgps.f2p3enc.ee?timestamp=1440706775
  Sending update to watch (31088 bytes)...
0000: 62123307430407df081b00011a6ed6fa00000000
0014: 0000013c7cd00d03fffe3107fffb06bb06bb06bb
0028: 06bb06bb06bb06bb06bb06bb06bb06bb06bb06bb
003c: 06bb06bb06bb06bb06bb67b6462000000000ffff
...
13c4: 03670000000c138a080000190095ecd21d47564f
13d8: c00003bd841c553a4304c987c045f1f595000004
13ec: 16080cd3ce096d66e5662e106c00301af412df24
at file position 0x13fe
fail: Resource temporarily unavailable
Failed to send QuickFixGPS update to watch.
```

It seems that the forced 20&nbsp;ms delay between packets in `tt_write_file` ([ttops.c#L207-L212](https://github.com/dlenski/ttblue/blob/f58df52e1441abe17216cb24395bceb3b9cb93b1/ttops.c#L207-L212)) is not sufficient for all devices, and I don't understand exactly how the Android app figures out the correct timing, though "trial-and-error like I'm doing" seems likely.

**Comment by dlenski** [comment]
Added `--gps-write-delay MICROSECONDS` option as de9ec129fbda9aace77707809dd12d5193698de7.

Does a longer delay, e.g. `--gps-write-delay 50000`, resolve this problem?

**Comment by weechatter** [comment]
It does not, i am sorry

**Comment by dlenski** [comment]
Thanks @weechatter ... you still get the exact same error with `--gps-write-delay 50000`?

**Comment by weechatter** [comment]
Error still occurs:
ttblue -D -D -a --gps-write-delay 50000
Updating QuickFixGPS...
  Saved 72 bytes to 00020001_20150907_195738.bin
  Last update was at Sat Sep  5 02:00:00 2015.  Downloading http://gpsquickfix.services.tomtom.com/fitness/sifgps.f2p3enc.ee?timestamp=1441648658
  Sending update to watch (32150 bytes)...
0000: 62123307450107df09070000e22db23500000000
...
13ec: 1909022e02113e29f02398038f00cae21b033619
at file position 0x13fe
fail: Resource temporarily unavailable
Failed to send QuickFixGPS update to watch.

**Comment by dlenski** [comment]
Darn. I'm at a loss here.

* I know the basic file-writing logic is correct because `ttblue` is able to set the correct "sync menu name" at the beginning. (**It shows the name of your computer in the sync menu, right?**)
* But something is going wrong when trying to write a long file that requires multiple checksums (they should occur every 256 packets = 256×20&nbsp;B = 0x1400&nbsp;B)

I think I would need to see an Android Bluetooth snooping log ([see here]( https://github.com/dlenski/ttblue/blob/master/tt_bluetooth.md#acknowledgments)) for your watch to see exactly how the official TomTom app communicates with your watch when sending the QuickFixGPS update :-/

**Comment by dlenski** [comment]
Hi @weechatter, I don't know if you're still using a TT watch, but I fixed some major bugs in 49b17050237bcff6bb519797011c3e2cffdb683b, which almost certainly prevented `ttblue` from working for anyone except me.

It would be great to know if it works for you now :+1:

**Comment by weechatter** [comment]
Hi Dan,
yes i am still using the TT watch. i am going to test tomorrow (if we don't have snow :-) )
